consul/resource_consul_service.go

```diff
@@ -375,6 +382,11 @@ func resourceConsulServiceRead(d *schema.ResourceData, meta interface{}) error {
 	sw.set("namespace", service.Namespace)
 	sw.set("partition", service.Partition)
 
+	sw.set("weights", map[string]int{
+		"passing": service.ServiceWeights.Passing,
+		"warning": service.ServiceWeights.Warning,
+	})
```

Comment:
We need to treat "" in state and "1" from Consul as equivalent, otherwise we'll always get a diff in the scenario where `weights` are not set.